pgjdbc/src/main/java/org/postgresql/core/v3/QueryExecutorImpl.java

```diff
@@ -2508,34 +2512,30 @@ private String receiveCommandStatus() throws IOException {
   }
 
   private void interpretCommandStatus(String status, ResultHandler handler) {
-    int update_count = 0;
-    long insert_oid = 0;
-
-    if (status.startsWith("INSERT") || status.startsWith("UPDATE") || status.startsWith("DELETE")
-        || status.startsWith("MOVE")) {
+    long oid = 0;
+    int count = 0;
+    Matcher matcher = COMMAND_COMPLETE_PATTERN.matcher(status);
+    if (matcher.matches()) {
+      // String command = matcher.group(1);
       try {
-        long updates = Long.parseLong(status.substring(1 + status.lastIndexOf(' ')));
-
-        // deal with situations where the update modifies more than 2^32 rows
-        if (updates > Integer.MAX_VALUE) {
-          update_count = Statement.SUCCESS_NO_INFO;
-        } else {
-          update_count = (int) updates;
-        }
-
-        if (status.startsWith("INSERT")) {
-          insert_oid =
-              Long.parseLong(status.substring(1 + status.indexOf(' '), status.lastIndexOf(' ')));
+        if (matcher.group(3) != null) {
```

Comment:
Would you extract a variable for `matcher.group(...)`?

Reply:
Sure that'd probably save an internal substring on the Matcher.